packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -594,6 +594,14 @@ export class ConnectionManager implements IConnectionManager {
 					LogLevel.verbose,
 				);
 			} catch (origError: any) {
+				this.logger.sendTelemetryEvent(
+					{
+						eventName: "ConnectToDeltaStreamException",
```

Comment:
If `connectToDeltaStream` throws, it's guaranteed that `connection` is `undefined`.  If that's not the case (based on PR description), then something else is throwing.

Just looked at the code and `DocumentDeltaConnection.disposed` itself has an `assert` in it!!  `0x244`.  The other alternative is one of the logger calls is throwing... I don't think I've ever seen that though.

So I'd strongly suggest including the thrown error in the new verbose log below.  I'd be very curious to see if it's that assert, and if not what the stack is.

In fact, unless the error is one of the `NoDelta` cases, then it will either trigger Container Closed event, or DeltaConnectionFailureToConnect, so you should be able to see today what the error is.  Do we see `0x244` or either of those events in the logs?  Is it possible it's falling to the `NoDeltaStream` case?  But that should have no trouble disposing later (not sure what evidence you have about it never disposing as you mentioned)